extensions/wikia/ARecoveryEngine/ResourceLoaders/PageFair/ResourceLoaderAdEnginePageFairDetectionModule.php

```diff
@@ -0,0 +1,50 @@
+<?php
```

Comment:
I agree with designing the code in a different way, but I'm not sure about next inheritance... :/ for me it provides mess and makes classes not testable, especially with few levels of extend